lucene/benchmark/build.gradle

```diff
@@ -37,5 +37,121 @@ dependencies {
     exclude module: "xml-apis"
   })
 
+  runtimeOnly project(':lucene:analysis:icu')
+
   testImplementation project(':lucene:test-framework')
 }
+
+def tempDir = file("temp")
+def workDir = file("work")
+
+task run(type: JavaExec) {
+  description "Run a perf test (optional: -PtaskAlg=conf/your-algorithm-file -PmaxHeapSize=1G)"
+  main 'org.apache.lucene.benchmark.byTask.Benchmark'
+  classpath sourceSets.main.runtimeClasspath
+  // allow these to be specified on the CLI via -PtaskAlg=  for example
+  def taskAlg = propertyOrDefault('taskAlg', 'conf/micro-standard.alg')
+  args = [taskAlg]
+
+  maxHeapSize = propertyOrDefault('maxHeapSize', '1G')
+
+  String stdOutStr = propertyOrDefault('standardOutput', null)
```

Comment:
Just had a random thought that if you don't redirect to a file the process is piped between gradle (parent) and this may cause artificial slowdowns on buffers between processes... Don't know if this matters but an alternative design could create a temporary file (task class has a method for creating task-relative temporary files), redirect the output into that file (always) and only pipe it to the console at the end if stdOutStr is not defined. 

I really don't know how these benchmarks are used in practice but I wanted to signal a potential issue here.

Reply:
ehh; I'd prefer to keep this the way it is.  The code/scripts in the alg files generally don't print tons of output, so I don't think there's a perf interference concern.

Reply:
Sure.